packages/vaadin-grid/test/typings/grid.types.ts

```diff
@@ -4,75 +4,120 @@ import '../../src/vaadin-grid-selection-column';
-  assert<GridItem>(event.detail.value);
+  assertType<TestGridItem>(event.detail.value);
 });
 
 grid.addEventListener('cell-activate', (event) => {
-  assert<GridItemModel>(event.detail.model);
+  assertType<GridItemModel<TestGridItem>>(event.detail.model);
 });
 
 grid.addEventListener('column-reorder', (event) => {
-  assert<GridColumnElement[]>(event.detail.columns);
+  assertType<GridColumnElement<TestGridItem>[]>(event.detail.columns);
 });
 
 grid.addEventListener('column-resize', (event) => {
-  assert<GridColumnElement>(event.detail.resizedColumn);
+  assertType<GridColumnElement<TestGridItem>>(event.detail.resizedColumn);
 });
 
 grid.addEventListener('loading-changed', (event) => {
-  assert<boolean>(event.detail.value);
+  assertType<boolean>(event.detail.value);
 });
 
 grid.addEventListener('expanded-items-changed', (event) => {
-  assert<GridItem[]>(event.detail.value);
+  assertType<TestGridItem[]>(event.detail.value);
 });
 
 grid.addEventListener('selected-items-changed', (event) => {
-  assert<GridItem[]>(event.detail.value);
+  assertType<TestGridItem[]>(event.detail.value);
 });
 
 grid.addEventListener('grid-dragstart', (event) => {
-  assert<GridItem[]>(event.detail.draggedItems);
+  assertType<TestGridItem[]>(event.detail.draggedItems);
 });
 
 grid.addEventListener('grid-drop', (event) => {
-  assert<GridItem>(event.detail.dropTargetItem);
-  assert<GridDropLocation>(event.detail.dropLocation);
+  assertType<TestGridItem>(event.detail.dropTargetItem);
+  assertType<GridDropLocation>(event.detail.dropLocation);
 });
 
 const filter = document.createElement('vaadin-grid-filter');
 
 filter.addEventListener('value-changed', (event) => {
-  assert<string>(event.detail.value);
+  assertType<string>(event.detail.value);
 });
 
 const selectionColumn = document.createElement('vaadin-grid-selection-column');
 
 selectionColumn.addEventListener('select-all-changed', (event) => {
-  assert<boolean>(event.detail.value);
+  assertType<boolean>(event.detail.value);
 });
 
 const sorter = document.createElement('vaadin-grid-sorter');
 
 sorter.addEventListener('direction-changed', (event) => {
-  assert<GridSorterDirection>(event.detail.value);
+  assertType<GridSorterDirection>(event.detail.value);
 });
 
 const sortColumn = document.createElement('vaadin-grid-sort-column');
 
 sortColumn.addEventListener('direction-changed', (event) => {
-  assert<GridSorterDirection>(event.detail.value);
+  assertType<GridSorterDirection>(event.detail.value);
 });
 
 const treeToggle = document.createElement('vaadin-grid-tree-toggle');
 
 treeToggle.addEventListener('expanded-changed', (event) => {
-  assert<boolean>(event.detail.value);
+  assertType<boolean>(event.detail.value);
 });
+
+// Verify mixins are correctly extended
+assertType<ScrollerElement>(grid);
+assertType<ElementMixin>(grid);
+assertType<ThemableMixin>(grid);
+assertType<A11yMixin>(grid);
+assertType<ActiveItemMixin<TestGridItem>>(grid);
```

Comment:
Not sure if we want to do this, but it was little effort and verifies the expected API for elements. Can add this for other elements as well if we decide on it.

Reply:
Added type tests for remaining grid sub-components - at least for those that already had a type declaration. The Grid Pro is still missing some, but that should be a separate issue.